packages/atomic/src/components/atomic-facet-manager/atomic-facet-manager.tsx

```diff
@@ -26,30 +29,38 @@ interface FacetElement extends HTMLElement {
 export class AtomicFacetManager implements InitializableComponent {
   @InitializeBindings() public bindings!: Bindings;
   private facetManager!: FacetManager;
+  public searchStatus!: SearchStatus;
+
+  @BindStateToController('searchStatus')
+  @State()
+  private searchStatusState!: SearchStatusState;
 
   @Element() private host!: HTMLDivElement;
 
   @BindStateToController('facetManager', {
-    onUpdateCallbackMethod: 'onFacetManagerUpdate',
+    onUpdateCallbackMethod: 'sortFacets',
   })
   @State()
   public facetManagerState!: FacetManagerState;
   @State() public error!: Error;
 
   public initialize() {
+    this.searchStatus = buildSearchStatus(this.bindings.engine);
     this.facetManager = buildFacetManager(this.bindings.engine);
-  }
 
-  public onFacetManagerUpdate() {
-    this.sortFacets();
+    // An update has to be forced for the facets to be visually updated, without being interacted with.
+    this.bindings.i18n.on('languageChanged', this.sortFacets);
   }
 
-  private sortFacets() {
+  private sortFacets = () => {
+    if (!this.searchStatusState.firstSearchExecuted) {
```

Comment:
It's useless to reorder placeholder, and can cause rendering issue if we change the language right away, e.g.,

```
<script>
  (async () => {
    await customElements.whenDefined('atomic-search-interface');
    const searchInterface = document.querySelector('#search');
    searchInterface.language = 'fr';

    await searchInterface.initialize({
      accessToken: 'xx564559b1-0045-48e1-953c-3addd1ee4457',
      organizationId: 'searchuisamples',
    });

    searchInterface.executeFirstSearch();
  })();
</script>
```
Note that it's not the correct way of doing this as well, usually it would be to set the right prop on atomic-search-interface from the start, but we never know if our users will be able to know the language before hand.

Reply:
>can cause rendering issue if we change the language right away,

Can you describe the issue ? Could it happen for other components ? If so, is there a generic mitigation we could add ?

Reply:
I believe the root cause is the sorting, they is a period of time where facets are "moved" by being re-appended to the DOM. If we change the language at the same time, it kind of gets stuck at the placeholder phase.